src/mri/exp_mult_mex.jl

```diff
@@ -0,0 +1,162 @@
+        for j = 1:N
+            col[j] = e ^ (-u[j] * v[i])
+        end
+        # column of B computed
+        for j = 1:L
+            row = A[j, :]
+            D[j,i] = transpose(row) * col
+        end
+    end
+    return D
+end
+
+function exp_mult_mex_helper(A, u::Vector{Float64},  v::Vector{Complex{Float64}})
+    n = size(u)
+    n = n[1]
+    M = size(v)
+    M = M[1]
+    L, N = size(A)
+
+    D = zeros(ComplexF64, L, M)
+    col = Vector{Complex{Float64}}(undef, N)
+    e = MathConstants.e
+    for i = 1:M
+        # compute ith column of B
+        # B has M columns, N rows
+        for j = 1:N
+            col[j] = e ^ (-u[j] * v[i])
+        end
+        # column of B computed
+        for j = 1:L
+            row = A[j, :]
+            D[j,i] = transpose(row) * col
+        end
+    end
+    return D
+end
+
+function exp_mult_mex_helper(A, u::Vector{Float64},  v::Vector{Float64})
+    n = size(u)
+    n = n[1]
+    M = size(v)
+    M = M[1]
+    L, N = size(A)
+
+    D = zeros(ComplexF64, L, M)
+    col = Vector{Complex{Float64}}(undef, N)
+    e = MathConstants.e
+    for i = 1:M
+        # compute ith column of B
+        # B has M columns, N rows
+        for j = 1:N
+            col[j] = e ^ (-u[j] * v[i])
+        end
+        # column of B computed
+        for j = 1:L
+            row = A[j, :]
+            D[j,i] = transpose(row) * col
+        end
+    end
+    return D
+end
+
+function exp_mult_mex_helper(A, u::Vector{Complex{Float64}},  v::Vector{Complex{Float64}})
+    n = size(u)
+    n = n[1]
+    M = size(v)
+    M = M[1]
+    L, N = size(A)
+    println("result returned, but not supposed have both u and v complex")
+
+    D = zeros(Complex{Float64}, L, M)
+    col = Vector{Complex{Float64}}(undef, N)
+    e = MathConstants.e
+    for i = 1:M
+        # compute ith column of B
+        # B has M columns, N rows
+        for j = 1:N
+            col[j] = e ^ (-u[j] * v[i])
+        end
+        # column of B computed
+        for j = 1:L
+            row = A[j, :]
+            D[j,i] = transpose(row) * col
+        end
+    end
+    return D
+end
+
+"""
+    exp_mult_mex(test::Symbol)
+
+Test function for exp_mult_mex.
+Finds percent error of memory-efficient routine vs. normal routime
+in
+    test        Symbol  symbol to indicate test routine
+out
+    N/A
+"""
+function exp_mult_mex(test::Symbol)
+    L = 10
+    N = 30
+    M = 20
+    A = randn(N,L) + im * randn(N,L)
+    ur = randn(N)
+    ui = randn(N)
+    vr = randn(M)
+    vi = randn(M)
+    u = ur + im * ui
+    v = vr + im * vi
+
+    d1 = A' * exp.(-ur * transpose(v))
+    d2 = exp_mult_mex(A, ur, v);
+    println(max_percent_diff(d1, d2))
+
+    d1 = A' * exp.(-u * transpose(vr))
+    d2 = exp_mult_mex(A, u, vr);
+    println(max_percent_diff(d1, d2))
+    
+    d1 = A' * exp.(-ur * transpose(v))
+    ur = reshape(ur, 1, :)
+    v = reshape(v, 1, :)
+    d2 = exp_xform(transpose(A), v, ur, mode = :matrix)
+    println(max_percent_diff(d1, d2))
```

Comment:
test routine must return `true`
and should have some `@test` tests in it.
probably using `isapprox` instead of printing `max_percent_diff`